Ndev 3656 prepare for load test uniswap

romanova-natasha opened the pull request:
2 users, 1h40m

<img width="923" alt="Screenshot 2025-04-30 at 16 05 09" src="https://github.com/user-attachments/assets/40f10f4a-70d6-4382-9444-efaab513c63b" />

<img width="917" alt="Screenshot 2025-04-30 at 16 06 31" src="https://github.com/user-attachments/assets/216e36bc-cbb6-45b2-b947-0c7957c79036" />




5 users, 1h40m

<img width="1013" alt="Screenshot 2025-05-05 at 00 37 45" src="https://github.com/user-attachments/assets/cf5a9cf1-58bd-469a-b426-44cdb1e00b17" />

<img width="1019" alt="Screenshot 2025-05-05 at 00 39 23" src="https://github.com/user-attachments/assets/48e6f815-8444-4a6d-b57d-f33e3f0fdf0d" />


10 users, 1h
We observed some RPC drops at the beginning and end of the run. This may be related to network instability, but there were no failures or exceptions while handling scheduled transactions by the proxy
<img width="915" alt="Screenshot 2025-05-05 at 13 08 58" src="https://github.com/user-attachments/assets/9b385fbb-47d4-4d96-b88e-2b0d2a794545" />
<img width="923" alt="Screenshot 2025-05-05 at 13 08 40" src="https://github.com/user-attachments/assets/c30e0a39-98bf-4b75-a47e-af36a83d7193" />
<img width="918" alt="Screenshot 2025-05-05 at 13 09 08" src="https://github.com/user-attachments/assets/89a04168-49e5-4aa3-b43d-15872367f879" />

sshiganov committed:
Init commit

sshiganov committed:
Fixed Dockerfiles

sshiganov committed:
Fixed actions

sshiganov committed:
1. Moved locust_prepare.py to test_start in scheduled_tx.py
2. Added base class for sched_trx to perform shared resources allocation
3. Get rid of extra param in ScheduledTrxEstimateRequest (otherwise there was an error)

sshiganov committed:
Added minor fixes

sshiganov committed:
fixed local config

sshiganov committed:
fixed according to PR comments

sshiganov committed:
fixed according to PR comments

sshiganov committed:
fixes according PR comments

sshiganov committed:
Merge branch 'develop' into NDEV-3685-add-shared-resources

romanova-natasha committed:
add with-uniswap flag to update-contracts command

add with-uniswap flag to update-contracts command

romanova-natasha committed:
add uniswap contracts deploy and prepare pools for swap

add uniswap contracts deploy and prepare pools for swap

romanova-natasha committed:
resolve conflicts

resolve conflicts

romanova-natasha committed:
Merge remote-tracking branch 'origin/develop' into NDEV-3656-prepare-for-load-test-uniswap

# Conflicts:
#	loadtesting/proxy/tests/scheduled_tx.py

romanova-natasha committed:
fix use_bank error for local run

fix use_bank error for local run

romanova-natasha committed:
Merge remote-tracking branch 'origin/develop' into NDEV-3656-prepare-for-load-test-uniswap

romanova-natasha committed:
add token to deployer

add token to deployer

romanova-natasha committed:
fix mint and add deployer account to accounts

fix mint and add deployer account to accounts

romanova-natasha committed:
delete the second pool, minor refactoring

delete the second pool, minor refactoring

Commit:
add an approve step before scheduled swap txs

add an approve step before scheduled swap txs

Commit:
move uniswap scheduled tx load test to a separate file

move uniswap scheduled tx load test to a separate file

romanova-natasha committed:
Merge branch 'develop' into NDEV-3656-prepare-for-load-test-uniswap

Commit:
use the correct ERC20Wrapper

use the correct ERC20Wrapper

Commit:
reduce swap amount

reduce swap amount

kristinaNikolaevaa commented on clickfile.py:
```diff
@@ -75,6 +75,7 @@
 HOODIES_CHAINLINK_GITHUB_URL = "https://github.com/hoodieshq/chainlink-neon"
 PROXY_GITHUB_URL = f"https://api.github.com/repos/{DOCKER_HUB_ORG_NAME}/neon-proxy.py"
 FAUCET_GITHUB_URL = f"https://api.github.com/repos/{DOCKER_HUB_ORG_NAME}/neon-faucet"
+UNISWAP_V3_GIT = f"git@github.com:{DOCKER_HUB_ORG_NAME}/Uniswap-V3-NEON.git"
```
lets make it as for others parts by http (for ssh there are should be ssh github configuration)

romanova-natasha replied:
I did it because I got 403 from http, I'll ask devopses to help and then fix it

romanova-natasha replied:
fixed

kristinaNikolaevaa commented on clickfile.py:
```diff
@@ -472,6 +474,13 @@ def update_contracts(branch):
     )
 
+    if with_uniswap:
+        Path(EXTERNAL_CONTRACT_PATH / "uniswap-v3").mkdir(parents=True, exist_ok=True)
```
why don't you use update_contracts_from_git() ?

romanova-natasha replied:
fixed

kristinaNikolaevaa commented on loadtesting/proxy/tests/scheduled_tx_uniswap.py:
```diff
@@ -0,0 +1,361 @@
+    return math.floor(887272 / tick_spacing) * tick_spacing
+
+
+def fund_solana_account(evm_loader, solana_account, bank_account, network, network_object):
```
instead of network_object  you can check bank_account
if it exists - use it

kristinaNikolaevaa commented on loadtesting/proxy/tests/scheduled_tx_uniswap.py:
```diff
@@ -0,0 +1,361 @@
+            solana_account=solana_account,
+            mintable=True,
+            bank_account=bank_account,
+            account=eth_account,
```
eth_account can be == deployer
no need to have 2 accounts

kristinaNikolaevaa commented on loadtesting/proxy/tests/scheduled_tx_uniswap.py:
```diff
@@ -0,0 +1,361 @@
+                erc20.mint_tokens(signer=erc20.account, to_address=neon_user.checksum_address, amount=token_mint_amount)
+            )
+
+        for item in receipts:
```
why do you need 2 loops
you can check status in previous one

kristinaNikolaevaa commented on loadtesting/proxy/tests/scheduled_tx_uniswap.py:
```diff
@@ -0,0 +1,361 @@
+    LOG.info("Approve for swap router...")
+    receipts = []
+    receipts.append(tokens["TTA"].approve(deployer, swap_router.address, token_mint_amount))
+    receipts.append(tokens["TTB"].approve(deployer, swap_router.address, token_mint_amount))
```
I think you don't need these 2 approves as you do it inside scheduled trx by neon users

kristinaNikolaevaa commented on loadtesting/proxy/tests/scheduled_tx_uniswap.py:
```diff
@@ -0,0 +1,361 @@
+        )
+
+        trx_estimate_0 = ScheduledTrxEstimateRequest(
+            self.uniswap_neon_account.checksum_address, token_in.contract_address, data, child_transaction=hex(2)
```
child_transaction=2

romanova-natasha replied:
NDEV-3744

kristinaNikolaevaa commented on loadtesting/proxy/tests/scheduled_tx_uniswap.py:
```diff
@@ -0,0 +1,361 @@
+        trx_estimate_0 = ScheduledTrxEstimateRequest(
+            self.uniswap_neon_account.checksum_address, token_in.contract_address, data, child_transaction=hex(2)
+        )
+        trx_estimate_1 = ScheduledTrxEstimateRequest(
```
child_transaction=3

kristinaNikolaevaa commented on loadtesting/proxy/tests/scheduled_tx_uniswap.py:
```diff
@@ -0,0 +1,361 @@
+        trx_estimate_1 = ScheduledTrxEstimateRequest(
+            self.uniswap_neon_account.checksum_address, token_out.contract_address, data, child_transaction="0xFFFF"
+        )
+        trx_estimate_2 = ScheduledTrxEstimateRequest(
```
child_transaction="0xFFFF" if 2 swaps are undependent

kristinaNikolaevaa commented on loadtesting/proxy/tests/scheduled_tx_uniswap.py:
```diff
@@ -0,0 +1,361 @@
+        tree_acc_data.add_trx(trxs[0], 2, 0)
+        tree_acc_data.add_trx(trxs[1], 2, 0)
+        tree_acc_data.add_trx(trxs[2], 0xFFFF, 0)
+        tree_acc_data.add_trx(trxs[3], 0xFFFF, 0)
```
something wrong with child 
lets have a call

Commit:
fix review

fix review

romanova-natasha committed:
Merge branch 'develop' into NDEV-3656-prepare-for-load-test-uniswap

Commit:
fix: fund solana account when test is long

fix: fund solana account when test is long

Commit:
fix: update uniswap from git and replace init code hash for pool address

fix: update uniswap from git and replace init code hash for pool address

Commit:
rm unused var

rm unused var

Commit:
fix child txs tags

fix child txs tags

Commit:
fix: thread locking when transaction is not successful, add: comment about strange tags in estimate_scheduled cause of the bug

fix: thread locking when transaction is not successful, add: comment about strange tags in estimate_scheduled cause of the bug

romanova-natasha committed:
Merge branch 'develop' into NDEV-3656-prepare-for-load-test-uniswap

kristinaNikolaevaa commented on clickfile.py:
```diff
@@ -472,6 +473,38 @@ def update_contracts(branch):
+            update_npm=True,
+        )
+
+        pool_addr_path = (
```
As I know there is only 1 countract where we need the replacement 
Do you want to check all here?

romanova-natasha replied:
It is a path to contract "PoolAddress.sol", we replace hash there only

kristinaNikolaevaa commented on clickfile.py:
```diff
@@ -472,6 +473,38 @@ def update_contracts(branch):
+        replacements = [
+            (
+                b"0xe34f199b19b2b4f47f68442619d555527d244f78a3297ea89325f843f87b8b54",
+                b"0xfeca55d18a66e13a3b004f5ea1833d181be8e62d7ac64669f176c76b5a79fc9d",
```
add please some comment why we do it

romanova-natasha replied:
done

kristinaNikolaevaa commented on loadtesting/proxy/tests/scheduled_tx.py:
```diff
@@ -160,62 +163,69 @@ class ScheduledTxsIndependentTasksSet(BaseScheduledTxTaskSet):
+            self.web3_client_sol.send_all_scheduled_transactions(trxs)
+            for trx in trxs:
+                check_trx_is_success(self.web3_client_sol, self.evm_loader, trx.hash().hex(), timeout=240)
+        except BaseException:
```
what about to move it to separate methods? to base class for example

romanova-natasha replied:
removed

kristinaNikolaevaa commented on loadtesting/proxy/tests/scheduled_tx_uniswap.py:
```diff
@@ -0,0 +1,352 @@
+    return math.floor(887272 / tick_spacing) * tick_spacing
+
+
+def fund_solana_account(evm_loader, solana_account, bank_account, network):
```
it can be moved to base class

romanova-natasha replied:
we use it before we declare our test class with base class inheritance

Commit:
rm try/catch

rm try/catch

Commit:
add comments

add comments

Commit:
fix log

fix log

Commit:
Merge remote-tracking branch 'origin/develop' into NDEV-3656-prepare-for-load-test-uniswap

# Conflicts:
#	loadtesting/proxy/tests/scheduled_tx.py
